### ISSUE TITLE
Problem with selection option

### ISSUE THREAD

**Issue by leocotes** [boppreh/keyboard#175]
Hello guys! Well, i'm new from python and i've a problem with my code.

So, I've a screen options, to select between some options, but i would like to put a counter, if the user doesn't select anyone in X seconds, the application select the default option.

`def print_header():
    system('clear')

    print('-----------------------------------------')
    print('       Python select               v1.0  ')
    print('-----------------------------------------\n')
    print('Press Ctrl+C to kill the application :)\n')
    print('Select options:\n')
    print('1 - A')
    print('2 - B')
    print('3 - C')
    print('4 - D')
    print('5 - E')
    print('6 - EXIT\n')

def select():
    time = 60
    while True:
        try:
            print_header()
            print('Select a options in %d seconds!' % time)
            if keyboard.is_pressed('1'):
                server = [16438]
                return server
                break
            elif keyboard.is_pressed('2'):
                server = [3068]
                return server
                break
            elif keyboard.is_pressed('3'):
                server = [16322]
                return server
                break
            elif keyboard.is_pressed('4'):
                server = [16438, 3068, 16322]
                return server
                break
            elif keyboard.is_pressed('5'):
                server = []
                return server
            elif keyboard.is_pressed('6'):
                exit(1)
                break
            elif tempo == 0:
                server = [16438, 3068, 16322]
                return server
                break
            else:
                time = time -1
                sleep(1)
        except KeyboardInterrupt:
            print_header()

            print('Ctrl+C pressed. Application Killed')
            print('Thank you):):)')
            exit(1)`

With this code a need to press anytime the option until select one.
Is like: 1111111111111111111111111111111111111111111111111111111111111111
So, anyone could help me?

**Comment by boppreh** [comment]
Hi

First, some comments:

- Avoid `while True` loops; they will peg the CPU at 100% until the user selects an option. This package has a lot of functions that do that for you, and more efficiently.
- To get a selection from the user, try `keyboard.read_hotkey()`. It returns the next key (or combination) the user presses, as a string.
- If you wrap your code in triple backticks, Github will format with proper indentation.
- Note this library is for *global* key events. This is useful if you want to react to key presses even if your application is not in focus. If you don't need this, try the standard library functions, such as `input` or `getch`.

If you really need the full power of this library, and not just get console input from the user, I see two possible solutions.

One is to setup each option as a hotkey, which are executed asynchronously, and implement the timeout with a simple `sleep`, or `keyboard.call_later`. The code would look something like this:

```python
import keyboard
import time

accepted = False
def accept():
    global accepted
    accepted = True
    keyboard.remove_all_hotkeys()

def option1():
    accept()
    print('option1')
    # More code.

def option2():
    accept()
    print('option2')
    # More code.

keyboard.add_hotkey('1', option1)
keyboard.add_hotkey('2', option2)

time.sleep(10)

if not accepted:
    keyboard.remove_all_hotkeys()
    print('timed out')
```

The other option would be to use queues and/or threads, similar to to how `get_hotkey` is implemented (
https://github.com/boppreh/keyboard/blob/e63e54a956c7dcbc11ded11098e53460d0b1297f/keyboard/__init__.py#L936 ).

Unfortunately neither of which are very simple and easy for a beginner.

**Comment by leocotes** [comment]
Thank you boppreh, i try read_hotkey(), but this stop the code until the user put some key, and i want that if the user doesn't put anything the code doesn't stop and select default option

**Comment by boppreh** [comment]
Yes, unfortunately `read_hotkey` is only a replacement for the current code, not for the timeout you want. Have a try at the other two solutions, but please keep in mind this is not a trivial problem.
